### ISSUE TITLE
How to pass a switch python argument into input_signature when using tf.function?

### ISSUE THREAD

**Issue by LionQYW** [tensorflow/tensorflow#53307]
Hi, I’m using Tensorflow 2.7.0 currently. There is no any other configuration things related to my issue.

I’m trying to set a switch for a tf.function, like set the input_signature for the call method of a Karas Model, so I can use the tf.saved_model.save API to save it. But it seems like I can not pass a python argument into the input_signature.

For example, I built a Transformer translation Model, I set an argument named “istraining” for the call method of this Model. When istraining = True, the Model run train mode and use the “teacher forcing” pattern to decode target language input. When istraining = False,  the model run translating mode and use the “auto regression” pattern to generate target language.

But I can not find a way to pass the python True/False argument into the input_signature of the tf.function decorator.

My currently solution is set a translating class function and a training class function separately and run get_concrete_function for these two functions separately when saving them. 

Is there a way to integrate these two functions into the model’s call method, and use a python argument True/False to control it after I save the model and reload it?

More generally, there is a reserved argument named “training” in Karas Layers and Karas Model, which is a python argument. if I wanna decorate the call method of a Karas Model, how should I set the input signature for this “training” argument?

**Comment by sushreebarsa** [comment]
@LionQYW 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thanks!

**Comment by LionQYW** [comment]
Hi, I don’t think this is a pure Karas Problem. It is mainly about the tf.function and tf.saved_model API. And this is also a question for tf.Module class.
